amd-sev/src/main.rs

```diff
@@ -140,23 +144,33 @@ fn main() {
         rip: 0x1000,
         rflags: 0x2,
         ..Default::default()
-    }).unwrap();
+    })
+    .unwrap();
 
     loop {
         match cpu.run().unwrap() {
             Reason::Halt => break,
 
             Reason::Io(io) => match io {
                 ReasonIo::Out { port, data } => match port {
-                    0x03f8 => for b in data {
-                        unsafe { libc::putchar(*b as i32); }
-                    },
+                    0x03f8 => {
+                        for b in data {
+                            unsafe {
+                                libc::putchar(*b as i32);
+                            }
+                        }
+                    }
 
                     _ => panic!("Unexpected IO port!"),
                 },
 
                 _ => panic!("Unexpected IO!"),
             },
+            Reason::Mmio {
+                addr: _,
+                data: _,
+                read: _,
+            } => panic!("Reason::Mmio is unused here; however, the compiler will complain of a non-exhaustive match if it isn't covered. This is an empty response to prevent that error."),
```

Comment:
That's entertaining :smiley:

How about printing the generic case and printing its debug representation instead?

```suggestion
            reason => panic!("Unexpected reason: {:?}", reason),
```

(the suggestion won't work literally)

Reply:
Added in latest revision.